src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1171,6 +1209,9 @@ class SelectStatementSegment(ansi.SelectStatementSegment):
             Ref("LimitClauseSegment", optional=True),
             Ref("NamedWindowSegment", optional=True),
         ]
+    ).copy(
+        insert=[Ref("ForClauseSegment", optional=True)],
+        before=Ref("LimitClauseSegment", optional=True),
```

Comment:
The `ForClauseSegment` is in MySQL's `UnorderedSelectStatementSegment`'s match and parse grammars, but I only added it to Postgres's `SelectStatementSegment`'s parse grammar for my test cases to pass. 
https://github.com/sqlfluff/sqlfluff/blob/83636f24350483de31438441e050a8c54b61a187/src/sqlfluff/dialects/dialect_mysql.py#L1543-L1578
Let me know if I'm missing anything.